org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/GitBasedRepository.java

```diff
@@ -56,30 +57,34 @@
 	public GitBasedRepository(String repositoryLocation) throws IOException, NoWorkTreeException, GitAPIException {
 		super(repositoryLocation);
 		FileRepositoryBuilder builder = new FileRepositoryBuilder();
-		Repository gitRepo = builder.setWorkTree(this.repositoryRoot.toFile()).setMustExist(false).build();
-		if (!new File(this.determineRepositoryPath(repositoryLocation) + "/.git").exists()) {
-		    gitRepo.create();
+		this.gitRepo = builder.setWorkTree(this.repositoryRoot.toFile()).setMustExist(true).build();
+		if (!new File(this.determineRepositoryPath(repositoryLocation) + File.separator + ".git").exists()) {
+			this.gitRepo.create();
 		}
-		this.git = new Git(gitRepo);
+		this.git = new Git(this.gitRepo);
 		if (!this.git.status().call().isClean()) {
-            this.addCommit();
+			this.addCommit(null);
```

Comment:
Please add another method `addCommit` handling this case. Passing `null` is not a good style.

The new method should state "files changed externally" as commit message.